### PR TITLE
executor: refine StreamAggExec when child is empty

### PR DESCRIPTION
**Pull request by XuHuaiyu**
## What have you changed? (mandatory)

This PR refines the StreamAgg process when ChildExec returns empty.
This processing logic can be used both in the old and new evaluation framework.

## What are the type of the changes (mandatory)?

The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested (mandatory)?

the existing tests

## Does this PR affect documentation (docs/docs-cn) update? (optional)
no

## Refer to a related PR or issue link (optional)
#6952

### CODE REVIEW

**Comment by XuHuaiyu** [comment]
PTAL @zz-jason @winoros

**Comment by zz-jason** [comment]
add a test that uses the new framework?

**Comment by zz-jason** [comment]
LGTM

**Comment by zz-jason** [comment]
@XuHuaiyu sqllogical test failed.

File: executor/aggregate.go [hunk 1]
```diff
@@ -860,38 +863,37 @@ func (e *StreamAggExec) consumeGroupRows() error {
...
+
 	// No more data.
 	if e.childrenResults[0].NumRows() == 0 {
-		if e.hasData || len(e.GroupByItems) == 0 {
-			err := e.appendResult2Chunk(chk)
-			if err != nil {
-				return errors.Trace(err)
-			}
+		if !e.isChildReturnEmpty {
+			err = e.appendResult2Chunk(chk)
```
Comment (zhexuany) [inline executor/aggregate.go]: Why don't check the value of `err` here? If you have any particular reason to do so, is it better you add it here?

**Comment by zz-jason** [comment]
@XuHuaiyu any update?

File: executor/aggregate.go (hunk 1 above)
Comment (XuHuaiyu) [reply executor/aggregate.go]: err is checked in line 892

File: expression/aggregation/descriptor.go [hunk 2]
```diff
@@ -110,37 +110,64 @@ func (a *AggFuncDesc) typeInfer(ctx sessionctx.Context) {
...
+// If there is no matching row for the inner table of an outer join,
+// an aggregation function only involves constant and/or columns belongs to the inner table
+// will be set to the null value.
 // The input stands for the schema of Aggregation's child. If the function can't produce a default value, the second
 // return value will be false.
-//
-// According to MySQL, DefaultValue of the aggregation function can be tested as the following sql:
-//
+// e.g.
+// mysql> create table t(a int);
```
Comment (zz-jason) [inline expression/aggregation/descriptor.go]: Can we simplify the comment? This format is not very formal 😂

Comment (XuHuaiyu) [reply expression/aggregation/descriptor.go]: ok, I'll have a try

**Comment by XuHuaiyu** [comment]
PTAL @zz-jason @winoros

File: executor/aggregate.go (hunk 1 above)
Comment (zhexuany) [reply executor/aggregate.go]: got it.

File: expression/aggregation/descriptor.go [hunk 3]
```diff
@@ -110,39 +110,86 @@ func (a *AggFuncDesc) typeInfer(ctx sessionctx.Context) {
...
+// | t     | a       | int(11) |
+// +-------+---------+---------+
+//
+// Query: `select a, avg(a), sum(a), count(a), bit_xor(a), bit_or(a), bit_and(a), max(a), min(a), group_concat(a) from t;`
+// +------+--------+--------+----------+------------+-----------+----------------------+--------+--------+-----------------+
+// | a    | avg(a) | sum(a) | count(a) | bit_xor(a) | bit_or(a) | bit_and(a)           | max(a) | min(a) | group_concat(a) |
+// +------+--------+--------+----------+------------+-----------+----------------------+--------+--------+-----------------+
+// | NULL |   NULL |   NULL |        0 |          0 |         0 | 18446744073709551615 |   NULL |   NULL | NULL            |
+// +------+--------+--------+----------+------------+-----------+----------------------+--------+--------+-----------------+
+func (a *AggFuncDesc) GetDefaultValue() (v types.Datum) {
```
Comment (zz-jason) [inline expression/aggregation/descriptor.go]: Can this function be moved to the `Aggregation` interface, it seems like belonging to the execution layer.

**Comment by XuHuaiyu** [comment]
PTAL @winoros

**Pull request merged**